xonsh/shell.py

```diff
@@ -161,6 +161,8 @@ def push(self, line):
 
     def reset_buffer(self):
         """Resets the line buffer."""
+        cmd = ''.join(list(filter(('\n').__ne__, self.buffer)))  
```

Comment:
Also, you should probably use a lambda like `lamda x: x != '\n'` rather than relying on the magic method of str.